Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda

```diff
@@ -0,0 +1,125 @@
+{-# OPTIONS --safe --experimental-lossy-unification #-}
+module Cubical.Cohomology.EilenbergMacLane.Rings.Z2-properties where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.HLevels
+
+open import Cubical.Data.Unit
+open import Cubical.Data.Nat hiding (_+_)
+
+open import Cubical.Algebra.Group
+open import Cubical.Algebra.Group.Morphisms
+open import Cubical.Algebra.Group.Instances.IntMod
+open import Cubical.Algebra.AbGroup
+open import Cubical.Algebra.AbGroup.Instances.Unit
+open import Cubical.Algebra.DirectSum.DirectSumHIT.Base
+open import Cubical.Algebra.Ring
+open import Cubical.Algebra.CommRing
+open import Cubical.Algebra.CommRing.Instances.IntMod
+
+open import Cubical.Homotopy.EilenbergMacLane.Order2
+
+open import Cubical.Cohomology.EilenbergMacLane.Base
+open import Cubical.Cohomology.EilenbergMacLane.CupProduct
+open import Cubical.Cohomology.EilenbergMacLane.RingStructure
+
+open Iso
+open PlusBis
+open RingTheory
+open IsGroupHom
+
+
+module _
+  (X : Type ℓ-zero)
+  where
+
+  -- Pratical notations
+  H*ℤ/2 : Type ℓ-zero
+  H*ℤ/2 = H* ℤ/2Ring X
+
+  H*Rℤ/2 : Ring ℓ-zero
+  H*Rℤ/2 = H*R ℤ/2Ring X
+
+open CommRingStr (snd ℤ/2CommRing) using ()
+  renaming
+  ( 0r        to 0ℤ/2
+  ; 1r        to 1ℤ/2
+  ; _+_       to _+ℤ/2_
+  ; -_        to -ℤ/2_
+  ; _·_       to _·ℤ/2_
+  ; +Assoc    to +ℤ/2Assoc
+  ; +IdL      to +ℤ/2IdL
+  ; +IdR      to +ℤ/2IdR
+  ; +Comm     to +ℤ/2Comm
+  ; ·Assoc    to ·ℤ/2Assoc
+  ; ·IdL      to ·ℤ/2IdL
+  ; ·IdR      to ·ℤ/2IdR
+  ; ·DistR+   to ·ℤ/2DistR+
+  ; ·Comm     to ·ℤ/2Comm
+  ; is-set    to isSetℤ/2     )
+
+module _
+  {X : Type ℓ-zero}
+  where
+
+  -- This notation is needed as G'' shouldn't have been put implicit
```

Comment:
This is a bit strong maybe?;) It's not needed when working over arbitrary rings, which is why it's left out. Maybe we should just introduce notation for the cup product, like x ⌣[ R ] y

Reply:
I don't know, I think G should have been put explicitly everytime.
I needed it all the time or there is unification issues.
Here, I have just put it like this in a file just for Z/2.

Unless, we plan to compute more cohomology ring with a different ring, I don't think we should bother modifying anything else

Reply:
I like the `x ⌣[ R ] y` suggestions

Reply:
Also, I wonder if the unification issues could easily be fixed using instance arguments. Seems much nicer than having explicit arguments

Reply:
I don't understand what you want me to do ? 
Shouldn't this be done in Axel file rand reverberate everywhere ?